src/inertia.js

```diff
@@ -46,38 +54,36 @@ export default {
     this.hideModal()
     this.showProgressBar()
 
-    if (this.cancelToken) {
-      this.cancelToken.cancel(this.cancelToken)
+    if (this.abortController) {
+      this.abortController.abort()
     }
 
-    this.cancelToken = axios.CancelToken.source()
+    this.abortController = new AbortController()
 
-    return axios({
+    return window.fetch(url, {
       method: method,
-      url: url,
-      data: data,
-      cancelToken: this.cancelToken.token,
+      ...this.hasBody(method) ? { body: JSON.stringify(data) } : {},
+      signal: this.abortController.signal,
+      credentials: 'include',
       headers: {
+        'Content-Type': 'application/json;charset=UTF-8',
         'Accept': 'text/html, application/xhtml+xml',
         'X-Requested-With': 'XMLHttpRequest',
         'X-Inertia': true,
+        'X-Xsrf-Token': this.getCookieValue('XSRF-TOKEN'),
```

Comment:
Isn't this a bit too Laravel specific?

Reply:
It’s exactly what `axios` was doing before, as far as I can tell. `axios` has options for the cookie and header name to use for XSRF tokens, but those weren’t being explicitly set; `x-xsrf-token` (header) and `XSRF-TOKEN` (cookie) are the defaults. I guess we could expose it as an option?

Reply:
Oh, I didn't know this was a generic thing across frameworks. Suppose it doesn't hurt to have then!

Reply:
Would be nice if you could expose that as an option, yes. Some frameworks call it XSRF, some call it CSRF, so this isn't super generic.